main.go

```diff
@@ -278,3 +291,156 @@ func isDaemonSetReady(status *appsv1.DaemonSetStatus) bool {
 		status.DesiredNumberScheduled == status.NumberReady &&
 		status.NumberMisscheduled == 0
 }
+
+//nolint:funlen
+func ConfigurePrivilegedServiceAccount(namespace string) error {
+	aRole := rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleSaName,
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{{
+			APIGroups:     []string{"security.openshift.io"},
+			Resources:     []string{"securitycontextconstraints"},
+			ResourceNames: []string{"privileged"},
+			Verbs:         []string{"use"},
+		},
+		},
+	}
+
+	aRoleBinding := rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleSaName,
+			Namespace: namespace,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      roleSaName,
+			Namespace: namespace,
+		}},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     roleSaName,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	aServiceAccount := v1core.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceAccount",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleSaName,
+			Namespace: namespace,
+		},
+	}
+
+	// create role
+	_, err := daemonsetClient.K8sClient.RbacV1().Roles(namespace).Create(context.TODO(), &aRole, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating role, err=%s", err)
+	}
+
+	// create rolebinding
+	_, err = daemonsetClient.K8sClient.RbacV1().RoleBindings(namespace).Create(context.TODO(), &aRoleBinding, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating role bindings, err=%s", err)
+	}
+	// create service account
+	_, err = daemonsetClient.K8sClient.CoreV1().ServiceAccounts(namespace).Create(context.TODO(), &aServiceAccount, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating service account, err=%s", err)
+	}
+	return nil
+}
+
+func initNamespace(namespace string) (err error) {
+	// delete namespace if present
+	if namespaceIsPresent(namespace) {
```

Comment:
This entire namespace deletion could be put into one function, instead of two functions. 
- If namespace exists, delete and wait for delete.
- If not, return.

Reply:
fixed